lib/pure/random.nim

```diff
@@ -631,12 +631,7 @@ when not defined(nimscript) and not defined(standalone):
     ## * `initRand proc<#initRand,int64>`_ that accepts a seed for a new Rand state
     ## * `randomize proc<#randomize>`_ that initializes the default RNG using the current time
     ## * `randomize proc<#randomize,int64>`_ that accepts a seed for the default RNG
-    when defined(js):
-      let time = int64(times.epochTime() * 1000) and 0x7fff_ffff
-      result = initRand(time)
-    else:
-      let now = times.getTime()
-      result = initRand(convert(Seconds, Nanoseconds, now.toUnix) + now.nanosecond)
+    result = initRand(getCpuTicks())
```

Comment:
this constrains the random module to work only on platforms where high-resolution timers are available - given that cputicks depends on non-standard behavior, it severly limits the platforms where Nim can be used - how to init `rand` is not a performance-critical operation - in fact, it would be trivial to continue using standardised C API for this without any significant loss - `random` is not a cryptographic random source, it's a best-effort proposition upon which _no_ code that actually requires randomness should rely upon - the granularity doesn't not change the utility of the module for any use cases for which its use is appropriate.

Reply:
it doesn't; fallback code can always be added to `getCpuTicks` to return something similar to what std/monotimes returns. rdtsc is available on all x86 processors since the pentium, and other platforms that nim supports have equivalent instructions which can be wrapped by `getCpuTicks`, see google/benchmark code here https://github.com/google/benchmark/blob/v1.1.0/src/cycleclock.h#L116 which handles more platform than nim supports.

Reply:
> and other platforms that nim supports have equivalent instructions which can be wrapped by getCpuTicks

Do you mean it will be implemented in the future? Does it support ARM?